crates/chia-tools/src/bin/validate-blockchain-db.rs

```diff
@@ -129,10 +133,29 @@ features that are validated:
             "SELECT coin_name, coinbase, puzzle_hash, coin_parent, amount FROM coin_record WHERE confirmed_index == ?;",
         )
         .expect("failed to prepare SQL statement finding created coins");
+    let mut select_peak = connection
+        .prepare("SELECT hash FROM current_peak WHERE key == 0;")
+        .expect("failed to prepare SQL statement finding peak");
+
+    let mut peak_row = select_peak.query([]).expect("failed to query current peak");
+    let peak_hash = peak_row
+        .next()
+        .expect("missing peak")
+        .expect("missing peak")
+        .get::<_, [u8; 32]>(0)
+        .expect("missing peak");
 
     let mut prev_hash = constants.genesis_challenge;
     let mut prev_height: i64 = args.start as i64 - 1;
```

Comment:
the previous height is supposed to be -1 if we start from 0. I suppose I could rename it to `expected_height` (and compare it without the + 1 etc. ) so it can start at 0. I don't think it matters so much though.
If `args.start` is non-zero, the `prev_hash` is updated in the first iteration of the loop.